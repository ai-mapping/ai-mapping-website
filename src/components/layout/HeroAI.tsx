'use client';

import React from 'react';
import Image from 'next/image';
import { ArrowRight } from 'lucide-react';
import { cn } from '@/lib/cn';

export type HeroAIProps = {
  badge?: string;
  title: string;
  description: string;
  primaryButtonText?: string;
  secondaryButtonText?: string;
  onPrimaryButtonClick?: () => void;
  onSecondaryButtonClick?: () => void;
  backgroundImage?: string;
  overlayOpacity?: number; // 0-100
  overlayColor?: 'primary' | 'secondary' | 'accent' | 'background' | 'dark';
};

export const defaultHeroAIProps: HeroAIProps = {
  badge: 'New Framework',
  title: 'Content-First Frontend Development with AI',
  description: 'AI-Mapping revolutionizes how websites are built by letting your content shape the interface, not the other way around.',
  primaryButtonText: 'Get Started',
  secondaryButtonText: 'View on GitHub',
  backgroundImage: '/images/hero/hero-image.png',
  overlayOpacity: 85,
  overlayColor: 'background',
};

const HeroAI: React.FC<Partial<HeroAIProps>> = ({
  badge = defaultHeroAIProps.badge,
  title = defaultHeroAIProps.title,
  description = defaultHeroAIProps.description,
  primaryButtonText = defaultHeroAIProps.primaryButtonText,
  secondaryButtonText = defaultHeroAIProps.secondaryButtonText,
  onPrimaryButtonClick,
  onSecondaryButtonClick,
  backgroundImage = defaultHeroAIProps.backgroundImage,
  overlayOpacity = defaultHeroAIProps.overlayOpacity,
  overlayColor = defaultHeroAIProps.overlayColor,
}) => {
  // Function to determine the overlay color class based on the overlayColor prop
  const getOverlayColorClass = () => {
    switch (overlayColor) {
      case 'primary':
        return 'from-primary-dark via-primary to-primary-dark';
      case 'secondary':
        return 'from-secondary via-secondary to-secondary';
      case 'accent':
        return 'from-accent via-accent to-accent';
      case 'dark':
        return 'from-black via-black to-black';
      case 'background':
      default:
        return 'from-black via-background to-black';
    }
  };

  return (
    <section className="relative py-28 overflow-hidden w-full">
      {/* Background Image */}
      {backgroundImage && (
        <div className="absolute inset-0 z-0">
          <Image
            src={backgroundImage}
            alt="Hero background"
            fill
            priority
            style={{ objectFit: 'cover' }}
            className="w-full h-full"
          />
        </div>
      )}
      
      {/* Overlay with gradient */}
      <div 
        className={`absolute inset-0 bg-gradient-to-r ${getOverlayColorClass()} z-0`}
        style={{ opacity: (overlayOpacity ?? 85) / 100 }}
        data-testid="hero-overlay"
      ></div>
      
      {/* Content Container */}
      <div className="container mx-auto px-4 md:px-6 lg:px-8 relative z-10 w-full">
        <div className="flex flex-col items-center text-center max-w-5xl mx-auto">
          {badge && (
            <div className="mb-6 animate-pulse">
              <div className="bg-primary px-5 py-2 rounded-full text-base font-semibold text-black inline-flex items-center shadow-glow">
                <span>{badge}</span>
              </div>
            </div>
          )}
          
          <h1 className="text-5xl md:text-6xl lg:text-7xl font-extrabold my-10 bg-gradient-to-r from-primary via-secondary to-accent bg-clip-text text-transparent leading-tight tracking-tight">
            {title}
          </h1>
          
          <p className="text-xl md:text-2xl lg:text-3xl text-neutral mb-14 max-w-4xl font-light leading-relaxed">
            {description}
          </p>
          
          <div className="flex flex-col sm:flex-row gap-5 mt-4">
            {primaryButtonText && (
              <button 
                className={cn("bg-primary text-black font-bold text-lg px-8 py-4 rounded-lg hover:bg-primary-light transition-all duration-300 flex items-center shadow-md hover:shadow-lg")}
                onClick={onPrimaryButtonClick}
              >
                {primaryButtonText} <ArrowRight className="ml-3 h-6 w-6" />
              </button>
            )}
            
            {secondaryButtonText && (
              <button 
                className={cn("bg-transparent border-2 border-neutral-dark text-white font-bold text-lg px-8 py-4 rounded-lg hover:border-primary hover:text-primary transition-all duration-300 hover:shadow-glow")}
                onClick={onSecondaryButtonClick}
              >
                {secondaryButtonText}
              </button>
            )}
          </div>
        </div>
      </div>
      
      {/* Background glow effect */}
      <div className="absolute bottom-0 left-1/2 -translate-x-1/2 w-2/3 h-2 bg-primary blur-xl"></div>
    </section>
  );
};

export default HeroAI;