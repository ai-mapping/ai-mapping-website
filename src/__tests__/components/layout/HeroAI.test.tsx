import { render, screen, fireEvent } from '@testing-library/react';
import HeroAI, { defaultHeroAIProps } from '@/components/layout/HeroAI';

// Mock Next.js Image component
jest.mock('next/image', () => ({
  __esModule: true,
  default: (props: any) => {
    // eslint-disable-next-line @next/next/no-img-element
    return <img 
      src={props.src} 
      alt={props.alt} 
      className={props.className} 
      style={props.style}
      data-testid="mock-image"
    />;
  },
}));

describe('HeroAI Component', () => {
  it('renders with default props', () => {
    const { container } = render(<HeroAI />);
    
    expect(screen.getByText('New Framework')).toBeInTheDocument();
    expect(screen.getByText('Content-First Frontend Development with AI')).toBeInTheDocument();
    expect(screen.getByText('AI-Mapping revolutionizes how websites are built by letting your content shape the interface, not the other way around.')).toBeInTheDocument();
    expect(screen.getByText('Get Started')).toBeInTheDocument();
    expect(screen.getByText('View on GitHub')).toBeInTheDocument();
    expect(screen.getByTestId('mock-image')).toHaveAttribute('src', '/images/hero/hero-image.png');
    
    expect(container).toMatchSnapshot();
  });

  it('renders with custom props', () => {
    render(
      <HeroAI
        badge="Custom Badge"
        title="Custom Title"
        description="Custom description text"
        primaryButtonText="Custom Primary"
        secondaryButtonText="Custom Secondary"
        backgroundImage="/custom-image.jpg"
        overlayOpacity={50}
        overlayColor="primary"
      />
    );
    
    expect(screen.getByText('Custom Badge')).toBeInTheDocument();
    expect(screen.getByText('Custom Title')).toBeInTheDocument();
    expect(screen.getByText('Custom description text')).toBeInTheDocument();
    expect(screen.getByText('Custom Primary')).toBeInTheDocument();
    expect(screen.getByText('Custom Secondary')).toBeInTheDocument();
    expect(screen.getByTestId('mock-image')).toHaveAttribute('src', '/custom-image.jpg');
    
    // Check overlay style
    const overlay = screen.getByTestId('hero-overlay');
    expect(overlay).toHaveStyle('opacity: 0.5');
    expect(overlay).toHaveClass('from-primary-dark');
  });

  it('calls onPrimaryButtonClick when primary button is clicked', () => {
    const mockOnPrimaryButtonClick = jest.fn();
    render(<HeroAI onPrimaryButtonClick={mockOnPrimaryButtonClick} />);
    
    fireEvent.click(screen.getByText('Get Started'));
    expect(mockOnPrimaryButtonClick).toHaveBeenCalledTimes(1);
  });

  it('calls onSecondaryButtonClick when secondary button is clicked', () => {
    const mockOnSecondaryButtonClick = jest.fn();
    render(<HeroAI onSecondaryButtonClick={mockOnSecondaryButtonClick} />);
    
    fireEvent.click(screen.getByText('View on GitHub'));
    expect(mockOnSecondaryButtonClick).toHaveBeenCalledTimes(1);
  });

  it('renders without badge when badge prop is not provided', () => {
    render(<HeroAI badge="" />);
    
    expect(screen.queryByText('New Framework')).not.toBeInTheDocument();
  });

  it('renders without buttons when button text props are not provided', () => {
    render(
      <HeroAI 
        primaryButtonText="" 
        secondaryButtonText="" 
      />
    );
    
    expect(screen.queryByText('Get Started')).not.toBeInTheDocument();
    expect(screen.queryByText('View on GitHub')).not.toBeInTheDocument();
  });

  it('renders without background image when backgroundImage prop is empty', () => {
    render(<HeroAI backgroundImage="" />);
    
    expect(screen.queryByTestId('mock-image')).not.toBeInTheDocument();
  });
});