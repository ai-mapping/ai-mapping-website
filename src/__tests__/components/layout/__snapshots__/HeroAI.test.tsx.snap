// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroAI Component renders with default props 1`] = `
<div>
  <section
    class="relative py-28 overflow-hidden w-full"
  >
    <div
      class="absolute inset-0 z-0"
    >
      <img
        alt="Hero background"
        class="w-full h-full"
        data-testid="mock-image"
        src="/images/hero/hero-image.png"
        style="object-fit: cover;"
      />
    </div>
    <div
      class="absolute inset-0 bg-gradient-to-r from-black via-background to-black z-0"
      data-testid="hero-overlay"
      style="opacity: 0.85;"
    />
    <div
      class="container mx-auto px-4 md:px-6 lg:px-8 relative z-10 w-full"
    >
      <div
        class="flex flex-col items-center text-center max-w-5xl mx-auto"
      >
        <div
          class="mb-6 animate-pulse"
        >
          <div
            class="bg-primary px-5 py-2 rounded-full text-base font-semibold text-black inline-flex items-center shadow-glow"
          >
            <span>
              New Framework
            </span>
          </div>
        </div>
        <h1
          class="text-5xl md:text-6xl lg:text-7xl font-extrabold my-10 bg-gradient-to-r from-primary via-secondary to-accent bg-clip-text text-transparent leading-tight tracking-tight"
        >
          Content-First Frontend Development with AI
        </h1>
        <p
          class="text-xl md:text-2xl lg:text-3xl text-neutral mb-14 max-w-4xl font-light leading-relaxed"
        >
          AI-Mapping revolutionizes how websites are built by letting your content shape the interface, not the other way around.
        </p>
        <div
          class="flex flex-col sm:flex-row gap-5 mt-4"
        >
          <button
            class="bg-primary text-black font-bold text-lg px-8 py-4 rounded-lg hover:bg-primary-light transition-all duration-300 flex items-center shadow-md hover:shadow-lg"
          >
            Get Started
             
            <svg
              class="lucide lucide-arrow-right ml-3 h-6 w-6"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 12h14"
              />
              <path
                d="m12 5 7 7-7 7"
              />
            </svg>
          </button>
          <button
            class="bg-transparent border-2 border-neutral-dark text-white font-bold text-lg px-8 py-4 rounded-lg hover:border-primary hover:text-primary transition-all duration-300 hover:shadow-glow"
          >
            View on GitHub
          </button>
        </div>
      </div>
    </div>
    <div
      class="absolute bottom-0 left-1/2 -translate-x-1/2 w-2/3 h-2 bg-primary blur-xl"
    />
  </section>
</div>
`;
